HLTrigger/Timer/src/FastTimerService.cc

```diff
@@ -38,1425 +36,1603 @@
-
-  m_enable_timing_modules   = m_enable_timing_modules       or
-                              m_enable_dqm_bymodule         or
-                              m_enable_dqm_bymoduletype     or
-                              m_enable_dqm_bypath_total     or
-                              m_enable_dqm_bypath_overhead  or
-                              m_enable_dqm_bypath_details   or
-                              m_enable_dqm_bypath_counters  or
-                              m_enable_dqm_bypath_exclusive;
-
-  m_enable_timing_exclusive = m_enable_timing_exclusive     or
-                              m_enable_dqm_bypath_exclusive;
-
-
-  registry.watchPreallocate(        this, & FastTimerService::preallocate );
-  registry.watchPreModuleBeginJob(  this, & FastTimerService::preModuleBeginJob );
-  registry.watchPreGlobalBeginRun(  this, & FastTimerService::preGlobalBeginRun );
-  registry.watchPreStreamBeginRun(  this, & FastTimerService::preStreamBeginRun );
-  registry.watchPostStreamBeginRun( this, & FastTimerService::postStreamBeginRun );
-  registry.watchPostStreamEndRun(   this, & FastTimerService::postStreamEndRun );
-  registry.watchPostStreamBeginLumi(this, & FastTimerService::postStreamBeginLumi );
-  registry.watchPostStreamEndLumi(  this, & FastTimerService::postStreamEndLumi );
-  registry.watchPostEndJob(         this, & FastTimerService::postEndJob );
-  registry.watchPostGlobalEndRun(   this, & FastTimerService::postGlobalEndRun );
-  registry.watchPreSourceEvent(     this, & FastTimerService::preSourceEvent );
-  registry.watchPostSourceEvent(    this, & FastTimerService::postSourceEvent );
-  registry.watchPreEvent(           this, & FastTimerService::preEvent );
-  registry.watchPostEvent(          this, & FastTimerService::postEvent );
-  /* Disable until can handle more than one thread per event
-  // watch per-path events
-  registry.watchPrePathEvent(       this, & FastTimerService::prePathEvent );
-  registry.watchPostPathEvent(      this, & FastTimerService::postPathEvent );
-  // watch per-module events if enabled
-  if (m_enable_timing_modules) {
-    registry.watchPreModuleEvent(            this, & FastTimerService::preModuleEvent );
-    registry.watchPostModuleEvent(           this, & FastTimerService::postModuleEvent );
-    registry.watchPreModuleEventDelayedGet(  this, & FastTimerService::preModuleEventDelayedGet );
-    registry.watchPostModuleEventDelayedGet( this, & FastTimerService::postModuleEventDelayedGet );
-  }
-  */
-
-  // if requested, reserve plots for timing vs. lumisection
-  // there is no need to store the id, as it wil always be 0
-  if (m_enable_dqm_byls)
-    reserveLuminosityPlots("ls", "lumisection", "lumisection", config.getUntrackedParameter<uint32_t>("dqmLumiSectionsRange"), 1.);
+///////////////////////////////////////////////////////////////////////////////
 
+// convert any std::chrono::duration to milliseconds
+template <class Rep, class Period>
```

Comment:
How about putting these two functions in an anonymous namespace? Otherwise they go into the global namespace and could potentially clash with a function in another shared library.